### PR TITLE
ci: change required labels for issue triage

### PR DESCRIPTION
**Pull request by jelbourn**
Issue triage _currently_ requires a component to be set and one of the following to be true for an issue to be considered triaged:
* Marked as a bug _and_ has a severity _and_ has a frequency
* Mark as a feature
* Marked as a refactor
* Marked as a discussion
* Marked as "confusing"
* Marked as "use-case"

This PR changes the rules so that (in addition to the component), triage
requires:
* A priority label (P0 through P5)
* Marked as a feature
* Marked as a discussion

Triage may also apply additional, optional info labels to issues.

[This document outlines the new priority scheme](https://docs.google.com/document/d/1mN2zWsr1pxChSTHC7UkOgl4PhhuoFONtG_zcMWeqLwA/preview).

While this PR is focused on issue triage and not PR triage, I have
changed the PR section triage to remove reference to the "effort: *" and
"risk: *" labels. Looking through recent PRs, Kapunahele is the only
person applying these, so it's clear that this bit is no longer widely
practiced.

This is just one step in the always-ongoing process of managing GitHub
labels. More adjustments will come over time. In writing this PR, I have
already unearthed a few more areas that can be polished in follow-ups.

### CODE REVIEW

File: docs/TRIAGE_AND_LABELS.md [hunk 1]
```diff
@@ -48,84 +49,80 @@ We will treat them as a component even thought no specific source tree is associ
...
+issue or PR.
 
+## Caretaker Triage Process (Initial Triage)
 
-### Type
+The caretaker assigns `comp:  *` labels to new issues as they come in.
+Untriaged issues can be found by selecting the issues with no milestone.
 
-What kind of problem is this?
+If an issue or PR obviously relates to a release regression, the caretaker must assigning an
```
Comment (petebacondarwin) [inline docs/TRIAGE_AND_LABELS.md:61]:
```suggestion
If an issue or PR obviously relates to a release regression, the caretaker must assign an
```

File: docs/TRIAGE_AND_LABELS.md [hunk 2]
```diff
@@ -48,84 +49,80 @@ We will treat them as a component even thought no specific source tree is associ
...
+Detailed triage can be done by anyone familiar with the issue subject matter.
 
-If it's obvious that the issue or PR is related to a release regression, the caretaker is also responsible for assigning the `severity(5): regression` label to make the issue or PR highly visible.
+### Step 1: Does the issue have enough information?
 
-The primary triage should be done on a daily basis so that the issues become available for secondary triage without much of delay.
+Gauge whether the issue has enough information to act upon. This typically includes a test case
+via StackBlitz or GitHub and steps to reproduce. If the issue may be legitimate but needs more
+information, add the "needs clarification" label. These labels can be revisted if the author can
+provide further clarification. If the issue does have enough enformation, move on to step 2.
```
Comment (petebacondarwin) [inline docs/TRIAGE_AND_LABELS.md:78]:
```suggestion
provide further clarification. If the issue does have enough information, move on to step 2.
```

File: docs/TRIAGE_AND_LABELS.md [hunk 3]
```diff
@@ -48,84 +49,80 @@ We will treat them as a component even thought no specific source tree is associ
...
 
-The reason why we limit the responsibility of the caretaker to this one label is that it is likely that without domain knowledge the caretaker could mislabel issues or lack knowledge of duplicate issues.
+### Step 2: Bug, feature, or discussion?
 
-Once the primary triage is done, the ng-bot automatically adds the milestone `needsTriage`.
+By default, all issues are considered bugs. Bug reports require only a priority label.
 
+If the issue is a feature request, apply the "feature" label. Use your judgement to determine
```
Comment (petebacondarwin) [inline docs/TRIAGE_AND_LABELS.md:84]:
```suggestion
If the issue is a feature request, apply the `type: feature` label. Use your judgement to determine
```

File: docs/TRIAGE_AND_LABELS.md [hunk 4]
```diff
@@ -48,84 +49,80 @@ We will treat them as a component even thought no specific source tree is associ
...
+whether the feature request is reasonable. If it's clear that the issue requests something
+infeasible, close the issue with a comment explaining why.
 
-## Component's owner Triage Process
+If the issue is an RFC or discussion, apply the "discussion" label. Use your judgement to determine
```
Comment (petebacondarwin) [inline docs/TRIAGE_AND_LABELS.md:88]:
```suggestion
If the issue is an RFC or discussion, apply the `type: RFC / Discussion / question` label. Use your judgement to determine
```

File: docs/TRIAGE_AND_LABELS.md [hunk 5]
```diff
@@ -48,84 +49,80 @@ We will treat them as a component even thought no specific source tree is associ
...
+| P3 | An issue that is relevant to core functions, but does not impede progress. Important, but not urgent.                                                                                                                                                |
+| P4 | A relatively minor issue that is not relevant to core functions, or relates only to the attractiveness or pleasantness of use of the system. Good to have but not necessary changes/fixes.                                                           |
+| P5 | The team acknowledges the request but (due to any number of reasons) does not plan to work on or accept contributions for this request. The issue remains open for discussion.                                                                       |
 
-The issues with type `type: feature`, `type: refactor` and `type: RFC / Discussion / question` do not require a frequency and severity.
 
-Following is the definition of various frequency and severity levels:
+Issues marked with "feature" or "discussion" don't require a priority.
```
Comment (petebacondarwin) [inline docs/TRIAGE_AND_LABELS.md:107]:
```suggestion
Issues marked with `type: feature` or `type: RFC / Discussion / question` don't require a priority. 
```

File: docs/TRIAGE_AND_LABELS.md [hunk 6]
```diff
@@ -48,84 +49,80 @@ We will treat them as a component even thought no specific source tree is associ
...
+### Step 4: Apply additional information labels
 
+Many optional labels provide additional context for issues. Consider adding any of the following if
+they apply to the issue:
 
-These criteria are then used to calculate a "user pain" score as follows:
+* Browser or operating system labels (`windows`, `ie11`, etc.)
+* Labels that inform the severity (`regression`, `has workaround`, `no workaround`)
+* Labels that categorize the bug (`performance`, `refactoring`, `memory leak`)
+* Community engagement labels (`good first issue`)
```
Comment (petebacondarwin) [inline docs/TRIAGE_AND_LABELS.md:117]: Are these the exact names of the labels? Or are they things like `os: windows`, `browser: ie11`, etc?

**Review by petebacondarwin — COMMENTED** [review]
The CI lint is not happy with the commit message.

File: docs/TRIAGE_AND_LABELS.md (hunk 3 above)
Comment (jelbourn) [reply docs/TRIAGE_AND_LABELS.md:84]: The linked document proposes renaming this to "feature" (eliminating the "type" prefix)

File: docs/TRIAGE_AND_LABELS.md (hunk 4 above)
Comment (jelbourn) [reply docs/TRIAGE_AND_LABELS.md:88]: The linked document proposes renaming this to "discussion" (eliminating the "type" prefix and shortening)

File: docs/TRIAGE_AND_LABELS.md (hunk 5 above)
Comment (jelbourn) [reply docs/TRIAGE_AND_LABELS.md:107]: Same as above

File: docs/TRIAGE_AND_LABELS.md (hunk 6 above)
Comment (jelbourn) [reply docs/TRIAGE_AND_LABELS.md:117]: These names should reflect the new names outlined in the linked document

Comment (petebacondarwin) [reply docs/TRIAGE_AND_LABELS.md:117]: In that case the angular-robot file should reflect the new label names too, yes?

File: docs/TRIAGE_AND_LABELS.md [hunk 7]
```diff
@@ -48,84 +49,80 @@ We will treat them as a component even thought no specific source tree is associ
...
 
-* `type: RFC / discussion / question`
-* `type: bug`
-* `type: docs`
-* `type: feature`
-* `type: performance`
-* `type: refactor`
-* `type: use-case`
-* `type: confusing`
+Primary triage should occur daily so that issues can move into secondary triage.
```
Comment (josephperrott) [inline docs/TRIAGE_AND_LABELS.md:65]: Should the terms used here be "Initial Triage" and "Detailed Triage"?

File: docs/TRIAGE_AND_LABELS.md [hunk 8]
```diff
@@ -48,84 +49,80 @@ We will treat them as a component even thought no specific source tree is associ
...
+| P3 | An issue that is relevant to core functions, but does not impede progress. Important, but not urgent.                                                                                                                                                |
+| P4 | A relatively minor issue that is not relevant to core functions, or relates only to the attractiveness or pleasantness of use of the system. Good to have but not necessary changes/fixes.                                                           |
+| P5 | The team acknowledges the request but (due to any number of reasons) does not plan to work on or accept contributions for this request. The issue remains open for discussion.                                                                       |
 
-The issues with type `type: feature`, `type: refactor` and `type: RFC / Discussion / question` do not require a frequency and severity.
 
-Following is the definition of various frequency and severity levels:
+Issues marked with "feature" or "discussion" don't require a priority.
```
Comment (josephperrott) [inline docs/TRIAGE_AND_LABELS.md:107]: Do they not require a priority are are not supposed to have a priority?  I would expect its the latter, but I am not sure.

File: docs/TRIAGE_AND_LABELS.md (hunk 6 above)
Comment (jelbourn) [reply docs/TRIAGE_AND_LABELS.md:117]: Aha, I did miss a couple spots. I think that gets them all.

File: docs/TRIAGE_AND_LABELS.md (hunk 8 above)
Comment (jelbourn) [reply docs/TRIAGE_AND_LABELS.md:107]: IMO not _require_ a priority, but may have one if it communicates some useful signal about the PR. We use priorities on components PRs to signal which ones the caretaker should prioritize.

File: docs/TRIAGE_AND_LABELS.md (hunk 7 above)
Comment (jelbourn) [reply docs/TRIAGE_AND_LABELS.md]: Updated the spots that I missed.

**Review by josephperrott — APPROVED** [review]
LGTM

**Pull request closed without merging**